src/App.tsx

```diff
@@ -17,18 +71,32 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                selectedOption={selectedOption}
+                onSelectChange={setSelectedOption}
+                query={query}
+                onQueryChange={setQuery}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {isLoading ? (
+                <Loader />
+              ) : (
+                <TodoList
+                  todos={filteredTodos}
+                  selectedTodoId={selectedTodo?.id}
+                  onSelect={setSelectedTodo}
+                />
```

Comment:
It seems like you're generating keys on render. This is not recommended as it can lead to performance issues. Keys should be stable between re-renders. Consider using unique identifiers as keys.